### PR TITLE
Created Entities

### PR DESCRIPTION
**Pull request by Slavka-JD**

### CODE REVIEW

**Comment by mushkin-v** [comment]
Do we need softDeletable field in this entity?

**Comment by Slavka-JD** [comment]
What for? When we'll add the possibility to delete Media, then we'll need it

File: src/AppBundle/Entity/Affiche.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+use AppBundle\Entity\Performance;
+
+/**
+ *
+ * @ORM\Table(name="affiche")
+ * @ORM\Entity
+ *
+ */
+class Affiche
+
```
Comment (spolischook) [inline src/AppBundle/Entity/Affiche.php]: code style, please

File: src/AppBundle/Entity/Affiche.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     *
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Performance", inversedBy="affiches", cascade={"persist"})
+     */
+    private $performance;
+    /**
```
Comment (spolischook) [inline src/AppBundle/Entity/Affiche.php]: empty line before this line is needed

File: src/AppBundle/Entity/User.php [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+     * @ORM\Column(type="string", length=45)
+     */
+    private $firstname;
+
+    /**
+     * @var string
+     * @Assert\NotBlank()
+     * @ORM\Column(type="string", length=45)
+     */
+    private $Lastname;
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]:
```
private $lastName;
```

File: src/AppBundle/Entity/Performance.php [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+namespace AppBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ *
+ * @ORM\Table(name="Performance")
```
Comment (spolischook) [inline src/AppBundle/Entity/Performance.php]: By convention we named table name in lower case
```
@ORM\Table(name="performance")
```

File: src/AppBundle/Entity/Performance.php [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    /**
+     * @var string
+     * @Assert\NotBlank()
+     * @ORM\Column(type="string", length=45)
+     */
+    private $title;
+
+    /**
+     * @var string
+     * @ORM\Column(type="string", length=45, nullable=true)
```
Comment (spolischook) [inline src/AppBundle/Entity/Performance.php]: length=45? Use type text for description

File: src/AppBundle/Entity/Performance.php [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+     * @ORM\Column(name="id", type="integer")
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var string
+     * @Assert\NotBlank()
+     * @ORM\Column(type="string", length=45)
```
Comment (spolischook) [inline src/AppBundle/Entity/Performance.php]: use 255 for length for all string values

File: src/AppBundle/Entity/Performance.php [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+     * @var string
+     * @ORM\Column(type="string", length=45, nullable=true)
+     */
+    private $description;
+
+    /**
+     * @var /Datetime
+     * @Assert\NotBlank()
+     * @ORM\Column(type="datetime")
+     *
```
Comment (spolischook) [inline src/AppBundle/Entity/Performance.php]: extra line

File: src/AppBundle/Entity/Performance.php [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+     */
+    private $premiere;
+
+//    add private $medias when MediaBundle is installed;
+
+    /**
+     * @var
```
Comment (spolischook) [inline src/AppBundle/Entity/Performance.php]: ?

File: src/AppBundle/Entity/Performance.php [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+     *
+     * @ORM\OneToMany(targetEntity="AppBundle\Entity\Affiche", mappedBy="performance", cascade={"persist"})
+     */
+    private $affiches;
+
+    /**
+     * @var
```
Comment (spolischook) [inline src/AppBundle/Entity/Performance.php]: ?

File: src/AppBundle/Entity/Performance.php [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+        return $this->premiere;
+    }
+
+    /**
+     * Set premiere
+     *
+     * @param \DateTime $premiere
+     * @return Performance
+     */
+    public function setPremiere($premiere)
```
Comment (spolischook) [inline src/AppBundle/Entity/Performance.php]: use type hinting

File: src/AppBundle/Entity/Performance.php [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+    {
+        $this->premiere = $premiere;
+        return $this;
```
Comment (spolischook) [inline src/AppBundle/Entity/Performance.php]: empty line before every "return" statement

File: src/AppBundle/Entity/Performance.php [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    /**
+     * Set affiche
+     * @param affiche $affiche
+     * @return Performance
+     */
+    public function setAffiche(Affiche $affiche)
+    {
+        $this->affiches[] = $affiche;
```
Comment (spolischook) [inline src/AppBundle/Entity/Performance.php]: in doc block 
```
@return Performance
```
but it return nothing

Comment (spolischook) [inline src/AppBundle/Entity/Performance.php]: need to return $this

File: src/AppBundle/Entity/Performance.php [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    /**
+     * Set role
+     * @param role $role
+     * @return Performance
+     */
+    public function setRole(Role $role)
+    {
+        $this->roles[] = $role;
```
Comment (spolischook) [inline src/AppBundle/Entity/Performance.php]: need to return $this

File: src/AppBundle/Entity/Role.php [hunk 14]
```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ *
```
Comment (spolischook) [inline src/AppBundle/Entity/Role.php]: extra line

File: src/AppBundle/Entity/Role.php [hunk 15]
```diff
@@ -0,0 +1,157 @@
...
+ * @ORM\Table(name="Role")
+ * @ORM\Entity
+ *
```
Comment (spolischook) [inline src/AppBundle/Entity/Role.php]: extra line

File: src/AppBundle/Entity/Role.php [hunk 16]
```diff
@@ -0,0 +1,157 @@
...
+
+namespace AppBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ *
+ * @ORM\Table(name="Role")
```
Comment (spolischook) [inline src/AppBundle/Entity/Role.php]: By convention we named table name in lower case in plural
```
@ORM\Table(name="roles")
```

File: src/AppBundle/Entity/Role.php [hunk 17]
```diff
@@ -0,0 +1,157 @@
...
+    private $role;
+
+    /**
+     * @var string
+     * @ORM\Column(type="string", length=45, nullable=true)
+     */
+    private $description;
+
+    /**
+     *
```
Comment (spolischook) [inline src/AppBundle/Entity/Role.php]: doc block is missing

File: src/AppBundle/Entity/Role.php [hunk 18]
```diff
@@ -0,0 +1,157 @@
...
+     * @ORM\Column(name="id", type="integer")
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var string
+     * @Assert\NotBlank()
```
Comment (spolischook) [inline src/AppBundle/Entity/Role.php]: empty line after "@var"

File: src/AppBundle/Entity/Role.php [hunk 19]
```diff
@@ -0,0 +1,157 @@
...
+     * @ORM\Column(type="string", length=45)
+     */
+    private $role;
+
+    /**
+     * @var string
```
Comment (spolischook) [inline src/AppBundle/Entity/Role.php]: empty line after "@var"

File: src/AppBundle/Entity/Role.php [hunk 20]
```diff
@@ -0,0 +1,157 @@
...
+    private $description;
+
+    /**
+     *
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Performance", inversedBy="roles", cascade={"persist"})
+     */
+    private $performance;
+
+    /**
+     * @var
```
Comment (spolischook) [inline src/AppBundle/Entity/Role.php]: empty line after "@var"
and fill what var this mean

File: src/AppBundle/Entity/Role.php [hunk 21]
```diff
@@ -0,0 +1,157 @@
...
+    /**
+     * Set role
+     *
+     * @param string $role
+     * @return Role
+     */
+    public function setRole($role)
+    {
+        $this->role = $role;
+        return $this;
```
Comment (spolischook) [inline src/AppBundle/Entity/Role.php]: empty line before $this

File: src/AppBundle/Entity/Role.php [hunk 22]
```diff
@@ -0,0 +1,157 @@
...
+    /**
+     * Set description
+     *
+     * @param string $description
+     * @return Performance
+     */
+    public function setDescription($description)
+    {
+        $this->description = $description;
+        return $this;
```
Comment (spolischook) [inline src/AppBundle/Entity/Role.php]: empty line before $this

File: src/AppBundle/Entity/Role.php [hunk 23]
```diff
@@ -0,0 +1,157 @@
...
+    /**
+     * Set performance
+     *
+     * @param Performance $performance
+     * @return Role
+     */
+    public function setPerformance(Performance $performance = null)
+    {
+        $this->performance = $performance;
+        return $this;
```
Comment (spolischook) [inline src/AppBundle/Entity/Role.php]: empty line before $this

File: src/AppBundle/Entity/Role.php [hunk 24]
```diff
@@ -0,0 +1,157 @@
...
+    /**
+     * Set user
+     *
+     * @param User $user
+     * @return Role
+     */
+    public function setUser(User $user = null)
+    {
+        $this->user = $user;
+        return $this;
```
Comment (spolischook) [inline src/AppBundle/Entity/Role.php]: empty line before $this

File: src/AppBundle/Entity/User.php [hunk 25]
```diff
@@ -0,0 +1,217 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ *
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: extra line

File: src/AppBundle/Entity/User.php [hunk 26]
```diff
@@ -0,0 +1,217 @@
...
+ * @ORM\Table(name="User")
+ * @ORM\Entity
+ *
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: extra line

File: src/AppBundle/Entity/User.php [hunk 27]
```diff
@@ -0,0 +1,217 @@
...
+
+namespace AppBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ *
+ * @ORM\Table(name="User")
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: By convention we named table name in lower case in plural
```
@ORM\Table(name="theatre_users")
```

File: src/AppBundle/Entity/User.php [hunk 28]
```diff
@@ -0,0 +1,217 @@
...
+     * @var integer
+     *
+     * @ORM\Column(name="id", type="integer")
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var string
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: empty line after "@var"

File: src/AppBundle/Entity/User.php [hunk 29]
```diff
@@ -0,0 +1,217 @@
...
+     * @Assert\NotBlank()
+     * @ORM\Column(type="string", length=45)
+     */
+    private $firstName;
+
+    /**
+     * @var string
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: empty line after "@var"

File: src/AppBundle/Entity/User.php [hunk 30]
```diff
@@ -0,0 +1,217 @@
...
+     * @Assert\NotBlank()
+     * @ORM\Column(type="string", length=45)
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: length=255

File: src/AppBundle/Entity/User.php [hunk 31]
```diff
@@ -0,0 +1,217 @@
...
+     */
+    private $lastName;
+
+    /**
+     * @var string
+     * @ORM\Column(type="string", length=45, nullable=true)
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: length=255

File: src/AppBundle/Entity/User.php [hunk 32]
```diff
@@ -0,0 +1,217 @@
...
+
+    /**
+     * @var string
+     * @Assert\NotBlank()
+     * @ORM\Column(type="string", length=45)
+     */
+    private $lastName;
+
+    /**
+     * @var string
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: empty line after "@var"

File: src/AppBundle/Entity/User.php [hunk 33]
```diff
@@ -0,0 +1,217 @@
...
+     * @ORM\Column(type="string", length=45, nullable=true)
+     */
+    private $middleName;
+
+    /**
+     * @var /Datetime
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: empty line after "@var"

File: src/AppBundle/Entity/User.php [hunk 34]
```diff
@@ -0,0 +1,217 @@
...
+     * @Assert\NotBlank()
+     * @ORM\Column(type="date")
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: @var /Datetime
@ORM\Column(type="date")
???

File: src/AppBundle/Entity/User.php [hunk 35]
```diff
@@ -0,0 +1,217 @@
...
+     *
+     */
+    private $dob;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(type="string", length=45)
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: length=255

File: src/AppBundle/Entity/User.php [hunk 36]
```diff
@@ -0,0 +1,217 @@
...
+    /**
+     * Set firstName
+     *
+     * @param string $firstName
+     * @return User
+     */
+    public function setFirstName($firstName)
+    {
+        $this->firstName = $firstName;
+        return $this;
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: empty line before $this

File: src/AppBundle/Entity/User.php [hunk 37]
```diff
@@ -0,0 +1,217 @@
...
+    /**
+     * Set lastName
+     *
+     * @param string $lastName
+     * @return User
+     */
+    public function setLastName($lastName)
+    {
+        $this->lastName = $lastName;
+        return $this;
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: empty line before $this

File: src/AppBundle/Entity/User.php [hunk 38]
```diff
@@ -0,0 +1,217 @@
...
+    /**
+     * Set middleName
+     *
+     * @param string $middleName
+     * @return User
+     */
+    public function setMiddleName($middleName)
+    {
+        $this->middleName = $middleName;
+        return $this;
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: empty line before $this

File: src/AppBundle/Entity/User.php [hunk 39]
```diff
@@ -0,0 +1,217 @@
...
+        return $this->dob;
+    }
+
+    /**
+     * Set dob
+     *
+     * @param \DateTime $dob
+     * @return User
+     */
+    public function setDob($dob)
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: type hinting

File: src/AppBundle/Entity/User.php [hunk 40]
```diff
@@ -0,0 +1,217 @@
...
+    {
+        $this->dob = $dob;
+        return $this;
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: empty line before $this

File: src/AppBundle/Entity/User.php [hunk 41]
```diff
@@ -0,0 +1,217 @@
...
+    /**
+     * Set position
+     *
+     * @param string $position
+     * @return User
+     */
+    public function setPosition($position)
+    {
+        $this->position = $position;
+        return $this;
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: empty line before $this

File: src/AppBundle/Entity/User.php [hunk 42]
```diff
@@ -0,0 +1,217 @@
...
+    }
+
+    /**
+     * Set role
+     * @param role $role
+     * @return User
+     */
+    public function setRole(Role $role)
+    {
+        $this->roles[] = $role;
```
Comment (spolischook) [inline src/AppBundle/Entity/User.php]: return needed

File: src/AppBundle/Entity/Affiche.php [hunk 43]
```diff
@@ -0,0 +1,103 @@
...
+     *
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Performance", inversedBy="affiches", cascade={"persist"})
+     */
+    private $performance;
+
+    /**
+     * @var /Datetime
+     *
+     * @Assert\NotBlank()
+     * @ORM\Column(type="datetime"
```
Comment (spolischook) [inline src/AppBundle/Entity/Affiche.php]: )

File: src/AppBundle/Entity/Affiche.php [hunk 44]
```diff
@@ -0,0 +1,103 @@
...
+        return $this->dateTime;
+    }
+
+    /**
+     * Set dateTime
+     *
+     * @param \DateTime $dateTime
+     * @return affiche
+     */
+    public function setDateTime($dateTime)
```
Comment (spolischook) [inline src/AppBundle/Entity/Affiche.php]: type hinting

File: src/AppBundle/Entity/Affiche.php [hunk 45]
```diff
@@ -0,0 +1,103 @@
...
+    public function getDateTime()
+    {
+        return $this->dateTime;
+    }
+
+    /**
+     * Set dateTime
+     *
+     * @param \DateTime $dateTime
+     * @return affiche
```
Comment (spolischook) [inline src/AppBundle/Entity/Affiche.php]: Affiche

**Pull request merged**